src/libs/Network.js

```diff
@@ -109,9 +109,13 @@ function processNetworkRequestQueue() {
         }
 
         // If we have a request then we need to check if it can be persisted in case we close the tab while offline
-        const retryableRequests = _.filter(networkRequestQueue, request => (
-            !request.data.doNotRetry && request.data.persist
-        ));
+        const retryableRequests = [];
+        _.each(networkRequestQueue, (request) => {
+            if ((request.data.doNotRetry || request.data.forceNetworkRequest) && !request.data.persist) {
+                return request.resolve();
```

Comment:
> I think this would maybe be clearer if we did
> 
> ```js
> request.resolve();
> return;
> ```
> 
> To show that the request is just being resolved but nothing is using the return value and we are just returning early out of the `each` iteration without retrying the request.

That does look better and provide more context!

Reply:
> It's also not very obvious why we are resolving it? Should we maybe add some context here?
> What happens if we just skip these requests as we were doing before? Can we explain it in plain terms?

Skipping the request makes `fetchCurrencyPreferences` get stuck [here](https://github.com/Expensify/Expensify.cash/blob/de88d27053659e880e1fb0dff05d94a55135d825/src/libs/actions/PersonalDetails.js#L246) because `API.GetPreferredCurrency` is unfulfilled and the rest of the code never runs. That means that we never set `isRetrievingCurrency = false` resulting in an infinite loading spinner. 

So, we can add the following comment as context: 
> Resolves requests that should not be retried (`doNotRetry` and `forceNetworkRequest`) and unblocks the rest of the call chain to be handled gracefully.

Reply:
Ok thanks! So I'm having a couple of thoughts here...

1. Why should we not retry this request? After trying to call it while offline - should we then call this method when we are online again so we can update the preferred currency?
2. Is there some way to make advancing to the next screen not have a dependency on this API request?

On the one hand, I think I understand that the original intention was to wait until we have the the preferred currency in hand and only after move to the next screen. But also wondering why this is something we need to "wait for". And curious if it would be better to...

1. Just move the user to the next screen and start off with a default currency
2. Get the preferred currency at some other point e.g. when the app inits so we can "trust" that it is accurate

Reply:
I like your idea of fetching the preferred currency when the app inits. I'm not sure there is a need to update it every time an IOU is initiated, as users most likely won't change countries/currencies that often. So, maybe we should:

1. Fetch preferred currency on app init. Already being done [here](https://github.com/Expensify/Expensify.cash/blob/de88d27053659e880e1fb0dff05d94a55135d825/src/libs/Navigation/AppNavigator/AuthScreens.js#L140)
2. Remove fetch currency calls from the IOU flow.
3. Revert `processNetworkRequestQueue` to its original logic.

What do you think?

Reply:
I think I like this plan! It more specifically addresses this problem and also establishes some patterns we are trying to develop with a lower possibility of introducing any surprising effects by messing with the lower level network code (although I think you might be onto something with the unresolved promises).

Perhaps before proceeding we can check to see when this code was added and by whom to be sure there was no particular reason to do things this way.

Seems like the changes were introduced here and reviewed by @Julesssss 
https://github.com/Expensify/Expensify.cash/pull/2194/files

How does this all sound @Julesssss ?

Reply:
I see that it was implemented [here](https://github.com/Expensify/Expensify.cash/pull/3514), so that users can make payments in different currencies but always have new IOUs default to their local currency. The solution proposed above would remove this feature. 

Another possible solution would be to introduce a temporary `iouCurrency` that defaults to the `preferredCurrency` but is not persisted across IOUs. That way users always have new IOUs start in their local currency, but can change it as necessary and we avoid the API call.

Thoughts?

Reply:
> so that users can make payments in different currencies but always have new IOUs default to their local currency

That's right, the original requirement was to make this check every time to account for users who arrive in a new country and should immediately see the local currency. If this requirement still holds then I think the use of a second key would be fine -- the choice to use a single `preferredCurrency` key for both **retrieved**, and **user-selected** currencies was just for simplicity.

CC @trjExpensify for additional context on the 'Always check for currency when opening IOU Modal' requirement

>  should we then call this method when we are online again so we can update the preferred currency?

@marcaaron does `NetworkConnection.onReconnect` provide the callback for this? If so, that seems like a good addition.

Reply:
@luacmartins maybe we can bring this conversation back to the original issue for more input. I don't have a great idea about how to solve this, but I don't think the solution lies in the network code changes.

Reply:
I agree that changing network is not a solution. However, implementing a second `selectedCurrency` key does seem like a viable simple option!